src/Symfony/Component/Cache/Traits/ContractsTrait.php

```diff
@@ -68,26 +69,27 @@ function (CacheItem $item, float $startTime, ?array &$metadata) {
             CacheItem::class
         );
 
-        return $this->contractsGet($pool, $key, function (CacheItem $item, bool &$save) use ($pool, $callback, $setMetadata, &$metadata) {
+        return $this->contractsGet($pool, $key, function (CacheItem $item, bool &$save) use ($pool, $callback, $setMetadata, &$metadata, $key) {
             // don't wrap nor save recursive calls
-            if (null === $callbackWrapper = $this->callbackWrapper) {
+            if (isset($this->computing[$key])) {
```

Comment:
is it expected that a value of null will be re-computed over and over, or should this be guarded against using array-key-exists?